src/binary/paths.ts

```diff
@@ -5,68 +5,103 @@ import {
   BINARY_UPDATE_URL,
   BINARY_UPDATE_VERSION_FILE_URL,
 } from "../globals/consts";
+import { ONPREM } from "../onPrem";
 
-let binaryRootPath: string | undefined;
+let BINARY_ROOT_PATH: string | undefined;
 const ARCHITECTURE = getArch();
 const SUFFIX = getSuffix();
 const BUNDLE_SUFFIX = getBundleSuffix();
 
 export async function setBinaryRootPath(
   extensionContext: vscode.ExtensionContext
 ): Promise<void> {
-  binaryRootPath =
+  if (ONPREM) {
+    const base = `binaries/${getArch()}-${getPlatform()}`;
+    BINARY_ROOT_PATH = path.join(extensionContext.extensionPath, base);
+    await makeExecutable(BINARY_ROOT_PATH);
```

Comment:
already implemented use  this https://github.com/codota/tabnine-vscode/blob/dc35e8a59c7ecd0e074c893fc0a58f5bfb3e1c20/src/binary/binaryFetcher/bundleDownloader.ts#L88